From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anonymous <no email>
Date: Wed, 10 Jun 2020 20:31:36 -0600
Subject: [PATCH] Add support for C++14 digit separators

See bug#58276
---
 src/cpp.lang | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/cpp.lang b/src/cpp.lang
index 480a393..f03e759 100644
--- a/src/cpp.lang
+++ b/src/cpp.lang
@@ -7,10 +7,12 @@ include "c_comment.lang"
 
 include "c_preprocessor.lang"
 
-include "number.lang"
-
 include "c_string.lang"
 
+number =
+'\<[+-]?((0b[01\']+)|(0x[[:xdigit:]\']+)|(([[:digit:]\']*\.)?
+[[:digit:]\']+([eE][+-]?[[:digit:]\']+)?))u?((int(?:8|16|32|64))|L)?\>'
+
 keyword = "__asm|__cdecl|__declspec|__export|__far16",
   "__fastcall|__fortran|__import",
   "__pascal|__rtti|__stdcall|_asm|_cdecl",
